client/src/main/java/org/mvndaemon/mvnd/client/DefaultClient.java

```diff
@@ -83,24 +84,20 @@ public static void main(String[] argv) throws Exception {
                 || Environment.COMPLETION.hasCommandLineOption(args);
 
         // Color
-        String styleColor = Environment.MAVEN_COLOR.getCommandLineOption(args);
-        if ("always".equals(styleColor) || "never".equals(styleColor)) {
-            /* If the user knows what he wants, pass his preference unchanged to the daemon */
-        } else if ("auto".equals(styleColor)) {
-            /* Do not pass auto to daemon, we handle it below */
-            Environment.MAVEN_COLOR.removeCommandLineOption(args);
-            styleColor = null;
-        } else if (styleColor != null) {
-            throw new IllegalArgumentException("Invalid color configuration option [" + styleColor
-                    + "]. Supported values are (auto|always|never).");
-        }
-
-        /* stdout is not a terminal e.g. when stdout is redirected to a file */
-        final boolean stdoutIsTerminal = CLibrary.isatty(1) != 0;
-        if (styleColor == null) {
-            Environment.MAVEN_COLOR.addCommandLineOption(
-                    args,
-                    (batchMode || logFile != null || !stdoutIsTerminal) ? "never" : "always");
+        final Color styleColor = Color.of(Environment.MAVEN_COLOR.getCommandLineOption(args))
+                .map(c -> {
+                    if (c == Color.auto) {
+                        /* Do not pass the auto value to the daemon, we translate it to either always or never below */
+                        Environment.MAVEN_COLOR.removeCommandLineOption(args);
+                    }
+                    return c;
+                }).orElse(Color.auto);
+
+        if (styleColor == Color.auto) {
+            /* stdout is not a terminal e.g. when stdout is redirected to a file */
+            final boolean stdoutIsTerminal = CLibrary.isatty(1) != 0;
+            args.add("-D" + Environment.MAVEN_COLOR.getProperty() + "="
+                    + ((batchMode || logFile != null || !stdoutIsTerminal) ? Color.never.name() : Color.always.name()));
```

Comment:
That looks a bit complicated.  Can we just remove the option from the command line
```
final Color styleColor = Color.of(Environment.MAVEN_COLOR.removeCommandLineOption(args)).orElse(Color.auto);
// translate auto to either always or never
if (style == Color.auto) {
    /* stdout is not a terminal e.g. when stdout is redirected to a file */
    final boolean stdoutIsTerminal = CLibrary.isatty(1) != 0;
    styleColor = (batchMode || logFile != null || !stdoutIsTerminal) ? Color.never : Color.always;
}
Environment.MAVEN_COLOR.addCommandLineOption(args, styleColor.name());
```

Reply:
+1, but we cannot call `Environment.MAVEN_COLOR.addCommandLineOption()` because that would send `--color` to the daemon, where on `-Dstyle.color` is supported

Reply:
Ok, could you add a comment to explain that ? That's not obvious...

Reply:
+1, done